clang/lib/Sema/SemaDecl.cpp

```diff
@@ -15841,13 +15841,24 @@ static void diagnoseImplicitlyRetainedSelf(Sema &S) {
           << FixItHint::CreateInsertion(P.first, "self->");
 }
 
+bool Sema::IsGetReturnObject(const FunctionDecl *FD) {
+  return isa<CXXMethodDecl>(FD) && FD->param_empty() &&
```

Comment:
NIT: maybe share the code between two functions? 
I am also not sure whether `isa<CXXMethodDecl>()` should also be added to the other check (aren't static functions also instances of `CXXMethodDecl`?)